packages/camera/camera/ios/Classes/FLTCam.m

```diff
@@ -220,11 +220,27 @@ - (void)captureToFile:(FLTThreadSafeFlutterResult *)result API_AVAILABLE(ios(10)
     return;
   }
 
-  [_capturePhotoOutput
-      capturePhotoWithSettings:settings
-                      delegate:[[FLTSavePhotoDelegate alloc] initWithPath:path
-                                                                   result:result
-                                                                  ioQueue:self.photoIOQueue]];
+  FLTSavePhotoDelegate *savePhotoDelegate = [[FLTSavePhotoDelegate alloc]
+           initWithPath:path
+                ioQueue:self.photoIOQueue
+      completionHandler:^(NSError *_Nullable error, NSString *_Nullable path) {
+        dispatch_async(self.captureSessionQueue, ^{
+          // Dispatch back to capture session queue to delete reference.
+          // Retain cycle is broken after the dictionary entry is cleared.
```

Comment:
Why do we need a retain cycle in the first place? Do we actually want to deliver the results still if the camera has been destroyed?

Reply:
This was to keep the original behavior with `selfReference`, which is deallocated only after completion, even after camera has been destroyed. We can change the behavior and use `weak self` here.

Reply:
I'm fine with preserving the behavior for now to minimize risk if we're not sure. It just seems odd to guarantee that callbacks would fine even on a deleted camera.